www/js/diary/infinite_scroll_detail.js

```diff
@@ -1,154 +1,73 @@
       const surveyOptKey = configObj.survey_info['trip-labels'];
       $scope.surveyOpt = SurveyOptions[surveyOptKey];
-      console.log('surveyOpt in infinite_scroll_details.js is', $scope.surveyOpt);
-      $scope.tripFilterFactory = $injector.get($scope.surveyOpt.filter);
-      $scope.filterInputs = $scope.tripFilterFactory.configuredFilters;
-    });
-  });
-
-  $scope.mapCtrl = {};
-  angular.extend($scope.mapCtrl, {
-    defaults : {
-    }
-  });
-
-  angular.extend($scope.mapCtrl.defaults, Config.getMapTiles())
-
-  var mapEvents = leafletMapEvents.getAvailableMapEvents();
-  for (var k in mapEvents) {
-    var eventName = 'leafletDirectiveMap.infscroll-detail.' + mapEvents[k];
-    $scope.$on(eventName, function(event, data){
-        try {
-            console.log("in mapEvents, event = "+JSON.stringify(event.name)+
-                  " leafletEvent = "+JSON.stringify(data.leafletEvent.type)+
-                  " leafletObject = "+JSON.stringify(data.leafletObject.getBounds()));
-        } catch (e) {
-            if (e instanceof TypeError) {
-                console.log("in mapEvents, event = "+JSON.stringify(event.name)+
-                      " leafletEvent = "+JSON.stringify(data.leafletEvent.type)+
-                      " leafletObject is undefined");
-            } else {
-                console.log(e);
-            }
-        }
-        $scope.eventDetected = event.name;
-    });
-  }
-
-  /*
-  leafletData.getMap('detail').then(function(map) {
-    map.on('touch', function(ev) {
-      alert("touch" + ev.latlng); // ev is an event object (MouseEvent in this case)
     });
   });
-  */
-
-  $scope.$on('leafletDirectiveMap.infscroll-detail.resize', function(event, data) {
-      console.log("diary/detail received resize event, invalidating map size");
-      data.leafletObject.invalidateSize();
-  });
-
-  $scope.refreshTiles = function() {
-      $scope.$broadcast('invalidateSize');
-  };
 
   $scope.trip = Timeline.getCompositeTrip($stateParams.tripId);
-  const tripgj = Timeline.compositeTrip2Geojson($scope.trip);
-  $scope.tripgj = $scope.trip;
-  $scope.tripgj.data = tripgj;
-  $scope.tripgj.common = {};
-  $scope.tripgj.common.earlierOrLater = '';
-  $scope.tripgj.pointToLayer = DiaryHelper.pointFormat;
+  $scope.formattedSectionProperties = DiaryHelper.getFormattedSectionProperties($scope.trip, ImperialConfig);
 
-  if (!angular.isDefined($scope.trip) || !angular.isDefined($scope.tripgj)) {
-    console.log("Detail trip = " + $scope.trip + " tripgj = " + $scope.tripgj + " not defined, going back to the list view")
+  if (!angular.isDefined($scope.trip)) {
+    console.log("Detail trip = " + $scope.trip + " not defined, going back to the list view")
     $state.go("root.main.inf_scroll");
-  }
-
-  $scope.recomputeDisplayTimelineEntries = function() {
-    console.log("Called inf scroll details.recomputeDisplayTimelineEntries");
-    const filterMap = $scope.filterInputs.map((f) => f.filter($scope.trip));
```

Comment:
The new implementation doesn't do this, but it can be adjusted for it
TODO

Reply:
Actually, what is the rationale behind this behavior? What have people said about it before?

To me the UX here seems sort of odd and unexpected. There's no warning that the redirect is going to happen, and with a 2 second delay, the timing is arbitrary and makes it seem like an accident, after the fact.

If I label a trip, and then stay on the details page for 2 seconds, chances are I probably intended to stay there; if I wanted to navigate away I would have done it already. So waiting 2 seconds and then navigating away seems like the most inconvenient thing you could do to the user.

I'd suggest either 1) remove the delay / make it very short (<0.5s); or 2) don't auto redirect; just update the main list and let the user go back to it once they're done looking around (would be my preference).

As a user and tester of the app, in my daily usage I don't actually use the details page when I label trips, but if I did I think this behavior would annoy me. I see the primary function of the details page not as labeling trips; rather getting more descriptive information about trips. So I might label a trip from there sometimes, but that doesn't mean I was done looking at it yet.

Reply:
The current behavior is that when a trip is labeled from the details screen, the view will stay on the details screen until the user navigates away.

For now, I am going to consider this the expected behavior. It seems correct to me; if we get any complaints about it we can re-evaluate.

Reply:
Sounds good. For the record, my husband frequently labels trips from the detail screen. He feels like the map in the label screen is too small, and goes into the detail screen so he can zoom in and see "exactly what happened" before labeling. I will see what he says about the new behavior once this gets to production.

I built in the "return to list" because it was easy to implement. But I've often felt that the right UX for the detail screen would be to support swiping - e.g. you are on the detail for trip n, you swipe right to see trip n+1, or swipe left to see trip n-1.

This would allow users to navigate through at a summary level (list view) or at a detail level (detail view) but without the awkward constant switching between them.

Reply:
Re: swiping between trips, I think this might be more feasible now that we are using the React Navigation architecture

But on the other hand, my competing intuition is that when you swipe to the right, you would expect to return the main label screen.